test/test_model.py

```diff
@@ -657,6 +657,13 @@ def test_storage(self):
                 self.model.storages.request('data', count_v)
 
     def test_relation_endpoint_bindings(self):
```

Comment:
This test feels like it has grown organically into something that is now a bit hard to actually understand. I'll try to grab the whole source of it, but I wonder if we can find a structure that would be a bit simpler.

Reply:
Looking at it inline makes me feel like it is a strong case for having a TestModelBindings that has its own setUp and helper methods rather than putting all of that context into local methods with closures.

Reply:
https://github.com/canonical/operator/pull/157/commits/d806429c7daeddd6d0d3c45f8646698fc015c760